rfcs/051-concepts-adapters/README.md

```diff
@@ -0,0 +1,60 @@
+# RFC 051: Ingesting Library of Congress concepts
+
+As per the high-level design of [RFC 052](https://github.com/wellcomecollection/docs/pull/83), one of the key areas of the concepts pipeline topology is an ingest section. This RFC outlines how this will be implemented for the first work phase (attaching only LoC concepts to works).
+
+![High level outline of the architecture in this RFC](https://user-images.githubusercontent.com/4429247/177178393-8fd615f1-f178-4a0a-83b2-a1d5ee09634f.png)
+
+### Retrieving concepts from bulk exports
+LoC provide bulk exports of all [subject headings](https://id.loc.gov/authorities/subjects.html) (LCSH) and [names](https://id.loc.gov/authorities/names.html) (LCNAF). These come in variety of flavours: in both MADS and SKOS vocabularies; and in N-Triple, JSON-LD, Turtle, and XML formats. All of the bulk exports are gzipped.
+
+We can re-run the ingest process on a schedule to capture the (infrequent) changes to LoC concepts, which are almost always additions rather than changes/deletions. 
+
+In the architecture diagram above, the "adapter" and "transformer" are outlined as separate services. In practice, we can perform this mini-ETL operation more efficiently and simply as a streaming operation within one service when we use the gzipped JSON-LD files; these express "one concept per line" and so we needn't worry about traversing any trees or indeed caring much about the underlying RDF.
+
+![A more detailed view of how a concepts ingestor might work](https://user-images.githubusercontent.com/4429247/177571629-56f7dd7b-9c43-4f38-bb11-aae53ed0c2e6.png)
+
+The "transform" step here is (a) the most ill-defined and (b) the only step that needs to differ for LCSH and LCNAF ingestors.
+
+### Transforming external concepts 
+
+[RFC 050](https://github.com/wellcomecollection/docs/tree/main/rfcs/050-concepts-api) outlined the rough design of a concepts API: obviously, we need to store the external concepts in a way that is sufficient to construct these API documents. We also don't want to extraneous data, as the majority of the concepts won't be used in the catalogue. 
+
+Based on this, we might transform the LCSH documents to something like:
+
+```json
+{
+  "id": "lc-subjects/sh12345678", // QName-type identifier 
+  "identifier": {
+    "value": "sh12345678",
+    "identifierType": "lc-subjects",
+    "type": "Identifier"
+  },
+  "label": "<value of skos:prefLabel>",
+  "alternativeLabels": [
+    "<values of skos:altLabel>"
+  ]
```

Comment:
I was going to suggest we record some sort of provenance information, e.g. the modified date of the LCSH dump we're using, for debugging purposes – but it's questionably useful, and screws with the upsert logic.

Let's leave that for now.

Reply:
Will note this down as I think it might be achievable